x-pack/plugins/profiling/public/services.ts

```diff
@@ -106,8 +102,7 @@ export function getServices(): Services {
         timeTo,
         kuery,
       };
-      const baseFlamegraph = (await http.get(paths.Flamechart, { query })) as BaseFlameGraph;
-      return createFlameGraph(baseFlamegraph);
```

Comment:
Unfortunately moving `createFlameGraph` inside the handler for the Kibana server endpoint will impact performance both by causing more CPU activity in the server but also increasing the payload size between the server and UI. In https://github.com/elastic/kibana/pull/142415, we split the flamegraph creation between the server and UI to address a prior performance regression.

Could we move `createFlameGraph` back into this method and the respective method that calls the new internal APM endpoint?

It does appear that the entire flamegraph creation pipeline will move inside the Elasticsearch plugin soon in order to improve the overall flamegraph latency. The split between server and UI will only have to continue until that move happens.

Reply:
Good catch. I moved it back here: https://github.com/elastic/kibana/pull/165360/commits/33c5ddbc8fca48132de16f0ef1a879774c907e62#diff-f83d94f2c6467652ae5c29e0a6bd49323fc62315c44bc1deec4e9392c1b1d51dR111